standard/navbar.lua

```diff
@@ -1,122 +1,127 @@
-local p = {}
-
-local getArgs
-
-function p._navbar(args)
-	local titleArg = 1
-
-	if args.collapsible then
-		titleArg = 2
-		if not args.plain then
-			args.mini = 1
-		end
-		if args.fontcolor then
-			args.fontstyle = 'color:' .. args.fontcolor .. ';'
-		end
-		args.style = 'float:left; text-align:left'
-	end
+local Class = require('Module:Class')
+local Logic = require('Module:Logic')
 
-	local titleText = args[titleArg] or (':' .. mw.getCurrentFrame():getParent():getTitle())
-	local title = mw.title.new(mw.text.trim(titleText), 'Template');
+local NavBar = {}
 
-	if not title then
-		error('Invalid title ' .. titleText)
-	end
+--legacy entry points
+function NavBar.navbar(args, headerText)
+	headerText = headerText or args[1]
+	return NavBar.NavBar(args, headerText)
+end
+function NavBar._navbar(args, headerText)
+	headerText = headerText or args[1]
+	return NavBar.NavBar(args, headerText)
+end
 
-	local talkpage = title.talkPageTitle and title.talkPageTitle.fullText or '';
+local _BRACKETS_LEFT = '&#91;'
+local _BRACKETS_RIGHT = '&#93;'
 
-	local div = mw.html.create():tag('div')
-	div
+function NavBar.NavBar(args, headerText)
+
+	local showBrackets = Logic.readBool(args.brackets)
+	local isCollapsible = Logic.readBool(args.collapsible)
+	local isPlain = Logic.readBool(args.plain)
+	local isMini = Logic.readBool(args.mini) or isCollapsible and (not isPlain)
+	local fontStyle = args.fontstyle
+	if isCollapsible and args.fontcolor then
+		fontStyle = 'color:' .. args.fontcolor .. ';'
+	end
+
+	local navBarDiv = mw.html.create('div')
 		:addClass('noprint')
 		:addClass('plainlinks')
 		:addClass('navbox-navbar')
-  		--:css('background', 'none')
-    	:css('padding', '0')
-      	:css('font-size', 'xx-small')
-		:cssText(args.style)
-
-	if args.mini then div:addClass('mini') end
-
-	if not (args.mini or args.plain) then
-		div
-			:tag('span')
-				:css('word-spacing', 0)
-				:cssText(args.fontstyle)
-				:wikitext(args.text or 'This box:')
-				:wikitext(' ')
+		:css('padding', '0')
+		:css('font-size', 'xx-small')
+	if isCollapsible then
+		navBarDiv:css('float', 'left'):css('text-align', 'left')
+	else
+		navBarDiv:cssText(args.style)
+	end
+	if isMini then
+		navBarDiv:addClass('mini')
+	end
+	if not isMini and not isPlain then
+		navBarDiv:node(mw.html.create('span')
+			:css('word-spacing', 0)
+			:cssText(fontStyle)
+			:wikitext(args.text or 'This box:')
+			:wikitext(' ')
+		)
 	end
 
-	if args.brackets then
-		div
-			:tag('span')
-				:css('margin-right', '-0.125em')
-				:cssText(args.fontstyle)
-				:wikitext('&#91; ')
+	local titleText = args.titleArg or (':' .. mw.getCurrentFrame():getParent():getTitle())
```

Comment:
Is this correct?

Reply:
If it is, it definitely needs a comment explaining itself.

Reply:
it was a straight copy paste from the current module

i adjusted it a bit and pushed it to a sep. function
hope that clears stuff already (it is the same procedure we also use for the `[h]` shortcuts in infoboxes